internal/v1/handler_compose_image.go

```diff
@@ -1256,5 +1279,74 @@ func (h *Handlers) buildCustomizations(ctx echo.Context, cr *ComposeRequest, d *
 		}
 	}
 
+	if aap := cr.ImageRequests[0].AAPRegistration; aap != nil {
+		tmpl := `#!/usr/bin/bash
+curl -s -i --data "host_config_key={{ .HostConfigKey }}" {{ .ServerAddress }}`
+
+		t, err := template.New("aap-script").Parse(tmpl)
```

Comment:
This could be extracted into a separate package together with templates. I did this on a different project and it worked well: https://github.com/foresterorg/forester/tree/main/internal/tmpl

Reply:
Cool, I've added a separate package and just reused most of what you had there